easybuild/main.py

```diff
@@ -590,10 +469,192 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
+                                 "Please use only one of them.")
+        else:
+            print_warning("Will not run the test step as requested via skip-test-step. "
+                          "Consider using ignore-test-failure instead and verify the results afterwards")
+
+    # determine easybuild-easyconfigs package install path
+    easyconfigs_pkg_paths = get_paths_for(subdir=EASYCONFIGS_PKG_SUBDIR)
+    if not easyconfigs_pkg_paths:
+        _log.warning("Failed to determine install path for easybuild-easyconfigs package.")
+
+    # if EasyStack file is provided, parse it, and loop over the items in the EasyStack file
+    if options.easystack:
+        orig_paths, opts_per_ec = parse_easystack(options.easystack)
+
+        _log.debug("Start build loop over items in the EasyStack file: %s" % orig_paths)
+        # TODO: insert fast loop that validates if all command line options are valid. If there are errors in options,
+        # we want to know early on, and this loop potentially builds a lot of packages and could take very long
+        # for path in orig_paths:
+        #     validate_command_opts(args, opts_per_ec[path])
+
+        # Loop over each item in the EasyStack file, each time updating the config
+        # This is because each item in an EasyStack file can have options associated with it
+        do_cleanup = True
+        for path in orig_paths:
+            _log.debug("Starting build for %s" % path)
+            # Whipe easyconfig caches
+            easyconfig._easyconfigs_cache.clear()
+            easyconfig._easyconfig_files_cache.clear()
+
+            # If EasyConfig specific arguments were supplied in EasyStack file
+            # merge arguments with original command line args
+            if path in opts_per_ec:
+                _log.debug("EasyConfig specific options have been specified for "
+                           "%s in the EasyStack file: %s" % (path, opts_per_ec[path]))
+                if args is None:
+                    args = sys.argv[1:]
+                ec_args = dict_to_argslist(opts_per_ec[path])
+                # By appending ec_args to args, ec_args take priority
+                new_args = args + ec_args
+                _log.info("Argument list for %s after merging command line arguments with EasyConfig specific "
+                          "options from the EasyStack file: %s" % (path, new_args))
+            else:
+                # If no EasyConfig specific arguments are defined, use original args.
+                # That way,set_up_configuration restores the original config
+                new_args = args
+            eb_go, cfg_settings = set_up_configuration(args=new_args, logfile=logfile, testing=testing,
+                                                       reconfigure=True)
+
+            hooks = load_hooks(options.hooks)
+            do_cleanup &= process_eb_args([path], eb_go.options, cfg_settings, modtool, testing, init_session_state,
```

Comment:
This should probably not use the modtool that is already setup, but set up a new one using
```
modtool = modules_tool(testing=testing)
```
In theory, the EasyStack file could specify to use Tcl modules for one EasyConfig, and Lmod for another.

Reply:
Exactly, and already fixed now.